tokio/src/runtime/basic_scheduler.rs

```diff
@@ -387,12 +326,14 @@ impl<P: Park> Drop for BasicScheduler<P> {
                 }
             }
 
-            assert!(context.shared.owned.is_empty());
+            assert!(context.spawner.shared.owned.is_empty());
```

Comment:
should this assertion maybe be 
```suggestion
            assert!(context.spawner.shared.owned.is_empty() || std::thread::panicking());
```
to avoid a double panic? we could maybe `eprintln!` a warning if we're panicking.

Reply:
Should totally double-panic / terminate the process. It is an internal bug that could have potentially bad results.

Really, we should consider aborting the process on internal bugs, but that is for another PR.

Reply:
I think we should probably be aborting the process on internal bugs, but i think we should _also_ make sure that the panic message (and the panic message from the internal bug) get printed before aborting, if we can.

Reply:
wdym? We still get panic messages from double panics?

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8dbaf07ea2606be49683478f945099e8

Reply:
oh, huh...i definitely have memories of that _not_ being the case? they must have fixed it at some point. never mind then!